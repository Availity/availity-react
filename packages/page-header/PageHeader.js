import React from 'react';
import PropTypes from 'prop-types';
import Breadcrumbs from '@availity/breadcrumbs';
import AppIcon from '@availity/app-icon';
import Feedback from '@availity/feedback';
import PayerLogo from '@availity/payer-logo';

const PageHeader = ({
  payerId,
  spaceName,
  spaceId,
  children,
  appName,
  appAbbr,
  iconColor,
  branded,
  crumbs,
  feedback,
  component,
  tag: Tag,
  ...props
}) => {
  if (spaceId || spaceName) {
    crumbs = [
      { name: spaceName, url: spaceId && `/web/spaces/spaces/#/${spaceId}` },
    ];
  }
  return (
    <React.Fragment>
      <div className="d-flex align-items-start">
        {crumbs && crumbs.type === Breadcrumbs ? (
          crumbs
        ) : (
          <Breadcrumbs crumbs={crumbs} active={appName || children} />
        )}
        {component}
      </div>
      <Tag className="page-header page-header-brand" {...props}>
        <div className="page-header-title">
          {!payerId && appAbbr && (
            <AppIcon color={iconColor} branded={branded} title={appName}>
              {appAbbr}
            </AppIcon>
          )}{' '}
          {children || appName}
        </div>
        {!payerId && feedback && (
          <Feedback
            appName={appName}
            className="float-md-right d-inline-block"
          />
        )}
        {payerId && (
          <PayerLogo
            spaceId={spaceId}
            payerId={payerId}
            className="float-md-right d-inline-block"
          />
        )}
      </Tag>
    </React.Fragment>
  );
};

PageHeader.propTypes = {
  appName: PropTypes.string.isRequired,
  spaceName: PropTypes.string,
  spaceId: PropTypes.string,
  appAbbr: PropTypes.string,
  iconColor: PropTypes.string,
  branded: PropTypes.bool,
  payerId: PropTypes.string,
  component: PropTypes.element,
  feedback: PropTypes.bool,
  children: PropTypes.node,
  crumbs: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
      })
    ),
    PropTypes.node,
  ]),
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
};

PageHeader.defaultProps = {
  tag: 'h1',
};

export default PageHeader;
