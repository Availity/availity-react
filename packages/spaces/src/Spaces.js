import React, { createContext, useContext, useReducer } from 'react';
import PropTypes from 'prop-types';
import { avSlotMachineApi } from '@availity/api-axios';
import { useEffectAsync } from '@availity/hooks';
import {
  spacesReducer,
  INITIAL_STATE,
  sanitizeSpaces,
  isFunction,
} from './helpers';

export const getAllSpaces = async (
  query,
  clientId,
  variables,
  _spaces = []
) => {
  if (!clientId) {
    throw new Error('clientId is required');
  }

  const {
    data: {
      data: { spaces },
    },
  } = await avSlotMachineApi.create(
    {
      query,
      variables,
    },
    { headers: { 'X-Client-ID': clientId } }
  );

  const { totalCount, page, perPage } = spaces;
  const unionedSpaces = _spaces.concat(spaces.spaces);

  if (totalCount > page * perPage) {
    const vars = { ...variables, page: page + 1 };
    return getAllSpaces(query, clientId, vars, unionedSpaces);
  }

  return unionedSpaces;
};

export const SpacesContext = createContext();

export const useSpacesContext = () => useContext(SpacesContext);

const Spaces = ({
  query,
  variables,
  clientId,
  spaceIds,
  payerIds,
  children,
  spaces: spacesFromProps,
}) => {
  const [{ spaces, loading, error }, dispatch] = useReducer(
    spacesReducer,
    INITIAL_STATE
  );

  // NOTE: we do not want to query slotmachine by payerIDs and spaceIDs at the same time
  // because slotmachine does an AND on those conditions. We want OR
  useEffectAsync(async () => {
    try {
      dispatch({
        type: 'LOADING',
        loading: true,
      });
      // Filter out dupes and ids that we already have the space for
      const filteredSpaceIDs = spaceIds
        .filter((id, i) => spaceIds.indexOf(id) === i)
        .filter(id => !spaces.some(spc => spc && spc.id === id))
        .filter(id => !spacesFromProps.some(spc => spc && spc.id === id));

      const filteredPayerIDs = payerIds
        .filter((id, i) => payerIds.indexOf(id) === i)
        .filter(
          id =>
            !spaces.some(
              spc => spc && spc.payerIDs && spc.payerIDs.some(pId => pId === id)
            )
        )
        .filter(
          id =>
            !spacesFromProps.some(
              spc => spc && spc.payerIDs && spc.payerIDs.some(pId => pId === id)
            )
        );

      if (filteredSpaceIDs.length === 0 && filteredPayerIDs.length === 0) {
        dispatch({
          type: 'LOADING',
          loading: false,
        });
        return;
      }

      let _spaces = [];
      if (filteredSpaceIDs.length > 0) {
        const vars = { ...variables, ids: filteredSpaceIDs };
        const spacesBySpaceIDs = await getAllSpaces(
          query,
          clientId,
          vars,
          spaces
        );
        _spaces = _spaces.concat(spacesBySpaceIDs);
      }

      if (filteredPayerIDs.length > 0) {
        const vars = { ...variables, payerIDs: filteredPayerIDs };
        const spacesByPayerIDs = await getAllSpaces(
          query,
          clientId,
          vars,
          spaces
        );
        _spaces = _spaces.concat(spacesByPayerIDs);
      }

      dispatch({
        type: 'SPACES',
        spaces: _spaces,
      });
    } catch (error_) {
      dispatch({
        type: 'ERROR',
        error: error_.message,
      });
    }
  }, [payerIds, spaceIds]);

  const spacesForProvider = sanitizeSpaces(spaces.concat(spacesFromProps));
  return (
    <SpacesContext.Provider
      value={{ spaces: spacesForProvider, loading, error }}
    >
      {isFunction(children)
        ? (() => children({ spaces: spacesForProvider, loading, error }))()
        : children}
    </SpacesContext.Provider>
  );
};

export const useSpaces = (...ids) => {
  const { spaces = [] } = useContext(SpacesContext) || {};

  const idsIsEmpty = !ids || ids.length === 0;
  const callerIsExpectingFirstSpace =
    ids && ids.length === 1 && ids[0] === undefined;

  if (callerIsExpectingFirstSpace && spaces.length > 1) {
    // eslint-disable-next-line no-console
    console.warn(
      `You did not pass an ID in to find a space, and there is more than 1 space in the space array. Returning all.`
    );
  }

  const shouldReturnAllSpaces = idsIsEmpty || callerIsExpectingFirstSpace;
  if (shouldReturnAllSpaces) {
    return spaces;
  }

  // Try to match by space id first, else match by payer id
  const filteredSpaces = ids.map(id => {
    let [spc] = spaces.filter(s => s.id === id);

    if (!spc) {
      [spc] = spaces.filter(s => {
        if (s.payerIDs && s.payerIDs.length > 0) return s.payerIDs[0] === id;
        return false;
      });
    }

    if (!spc) {
      [spc] = spaces.filter(s => (s.payerIDs || []).some(p => p === id));
    }
    return spc;
  });
  return filteredSpaces;
};

Spaces.propTypes = {
  clientId: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  query: PropTypes.string,
  variables: PropTypes.object,
  spaceIds: PropTypes.arrayOf(PropTypes.string),
  payerIds: PropTypes.arrayOf(PropTypes.string),
  spaces: PropTypes.arrayOf(PropTypes.object),
};

Spaces.defaultProps = {
  query: `
    query($ids: [String!], $payerIDs: [String!], $types: [String!], $page: Int){
      spaces(ids: $ids, payerIDs: $payerIDs, types: $types, page: $page){
        totalCount
        perPage
        page
        spaces{
          id
          name
          description
          link {
            url
          }
          payerIDs
          parentIDs
          metadata{
            name
            value
          }
          images{
            name
            value
          }
          url
        }
      }
    }
  `,
  variables: { types: ['space'] },
  spaceIds: [],
  payerIds: [],
  spaces: [],
};

export default Spaces;
