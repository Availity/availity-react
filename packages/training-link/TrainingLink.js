import React from 'react';
import PropTypes from 'prop-types';

const shouldProbablyBeInUIKit = {
  paddingTop: '12px',
};

const TrainingLink = ({ name, link }) => (
  <span style={shouldProbablyBeInUIKit} className="ml-auto">
    Need Help?{' '}
    <a href={link} target="_blank" rel="noopener noreferrer">
      Watch a demo
    </a>{' '}
    for {name}
  </span>
);

TrainingLink.propTypes = {
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default TrainingLink;
