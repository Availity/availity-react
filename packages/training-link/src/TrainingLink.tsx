import React from 'react';

const shouldProbablyBeInUIKit = { paddingTop: '12px' };
const linkStyles = { fontWeight: 'bold' } as const;

export type TrainingLinkProps = {
  name: string;
  link: string;
};

const TrainingLink = ({ name, link }: TrainingLinkProps): JSX.Element => (
  <span style={shouldProbablyBeInUIKit} className="ml-auto">
    Need Help?{' '}
    <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
      Watch a demo
    </a>{' '}
    <span className="sr-only">(opens in a new tab) </span>
    for {name}
  </span>
);

export default TrainingLink;
