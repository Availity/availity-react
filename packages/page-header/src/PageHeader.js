import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Breadcrumbs from '@availity/breadcrumbs';
import AppIcon from '@availity/app-icon';
import Feedback from '@availity/feedback';
import Spaces, { SpacesLogo, useSpaces, useSpacesContext } from '@availity/spaces';
import '../styles.scss';

const PageHeader = ({
  payerId,
  logo,
  spaceName,
  spaceId,
  children,
  appName,
  appAbbr,
  iconColor,
  branded,
  showCrumbs,
  crumbs,
  feedback: showFeedback,
  feedbackProps,
  titleProps: { className: titleClassName, ...restTitleProps },
  renderRightContent: RenderRightContent,
  renderRightClassName,
  component,
  tag: Tag,
  clientId,
  iconSrc,
  iconAlt,
  homeUrl,
  linkTag,
  className,
  variables,
  ...props
}) => {
  const spaces = useSpaces(spaceId, payerId);

  let firstSpaceForPayerID;
  let firstSpaceForSpaceID;
  if (!spaceId && payerId && spaces[0]) {
    firstSpaceForPayerID = spaces[0];
  } else {
    firstSpaceForSpaceID = spaces[0];
    firstSpaceForPayerID = spaces[1];
  }
  const { loading: spacesIsLoading } = useSpacesContext() || {};

  const _space = firstSpaceForSpaceID || firstSpaceForPayerID;

  let payerLogo = null;
  if (payerId || logo) {
    const logoAttrs = {
      spaceId,
      payerId,
      className: 'd-inline-flex',
      skeletonProps: {
        width: 180,
        height: '100%',
      },
    };
    payerLogo =
      firstSpaceForPayerID || (logo && firstSpaceForSpaceID) || spacesIsLoading ? (
        <SpacesLogo {...logoAttrs} />
      ) : (
        <Spaces
          spaceIds={spaceId ? [spaceId] : undefined}
          payerIds={payerId ? [payerId] : undefined}
          clientId={clientId}
          variables={variables}
        >
          <SpacesLogo {...logoAttrs} />
        </Spaces>
      );
  }

  const _spaceName = spaceName || (_space && _space.name);
  if ((spaceId || _spaceName) && !crumbs) {
    const url =
      firstSpaceForSpaceID && firstSpaceForSpaceID.link && firstSpaceForSpaceID.link.url
        ? firstSpaceForSpaceID.link.url
        : `/web/spc/spaces/#/${spaceId}`;
    crumbs = [{ name: _spaceName, url }];
  }

  const feedback = useMemo(() => <Feedback appName={appName} {...feedbackProps} />, [appName, feedbackProps]);

  return (
    <>
      <div className="page-header-above">
        {showCrumbs ? (
          <>
            {React.isValidElement(crumbs) ? (
              crumbs
            ) : (
              <Breadcrumbs crumbs={crumbs} active={appName || children} homeUrl={homeUrl} linkTag={linkTag} />
            )}
          </>
        ) : null}
        {component}
      </div>
      <div className={classNames('page-header', className)} data-testid="page-header" {...props}>
        <Tag
          className={classNames('page-header-title', titleClassName)}
          data-testid="page-header-title"
          {...restTitleProps}
        >
          {!payerId && appAbbr && (
            <AppIcon
              data-testid="page-header-app-icon"
              color={iconColor}
              branded={branded}
              title={appName}
              src={iconSrc}
              alt={iconAlt}
              aria-hidden
            >
              {appAbbr}
            </AppIcon>
          )}
          {children || <h1>{appName}</h1>}
        </Tag>
        {!RenderRightContent ? (
          <div className="page-header-right">
            {showFeedback && feedback}
            {payerLogo}
          </div>
        ) : (
          <RenderRightContent className={renderRightClassName} payerLogo={payerLogo} feedback={feedback} />
        )}
      </div>
    </>
  );
};

PageHeader.propTypes = {
  className: PropTypes.string,
  appName: PropTypes.string.isRequired,
  spaceName: PropTypes.string,
  spaceId: PropTypes.string,
  appAbbr: PropTypes.string,
  iconColor: PropTypes.string,
  branded: PropTypes.bool,
  payerId: PropTypes.string,
  component: PropTypes.element,
  feedback: PropTypes.bool,
  feedbackProps: PropTypes.shape({ ...Feedback.propTypes }),
  titleProps: PropTypes.object,
  children: PropTypes.node,
  linkTag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  showCrumbs: PropTypes.bool,
  crumbs: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        url: PropTypes.string,
      })
    ),
    PropTypes.node,
  ]),
  logo: PropTypes.bool,
  tag: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
  renderRightContent: PropTypes.func,
  renderRightClassName: PropTypes.string,
  homeUrl: PropTypes.string,
  clientId: PropTypes.string,
  iconSrc: PropTypes.string,
  iconAlt: PropTypes.string,
  variables: PropTypes.object,
};

PageHeader.defaultProps = {
  tag: 'div',
  // SpaceId and payerId are defaulted to null to prevent `useSpace` from
  // returning a space we may not want
  spaceId: null,
  payerId: null,
  homeUrl: '/public/apps/dashboard',
  titleProps: {},
  renderRightClassName: 'page-header-right',
  showCrumbs: true,
};

export default PageHeader;
