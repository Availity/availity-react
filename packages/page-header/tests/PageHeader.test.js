import React from 'react';
import { render, cleanup, waitForElement } from '@testing-library/react';
import TrainingLink from '@availity/training-link';
import { avSlotMachineApi } from '@availity/api-axios';
import { NavLink } from 'reactstrap';
import Spaces from '@availity/spaces';
import PageHeader from '..';

jest.mock('@availity/api-axios');

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe('PageHeader', () => {
  test('should render', () => {
    const { getByTestId } = render(<PageHeader appName="Payer Space" />);

    const pageHeaderTitle = getByTestId('page-header-title');

    expect(pageHeaderTitle.textContent).toEqual('Payer Space');
  });

  test('should render app icon', () => {
    const { getByText } = render(
      <PageHeader appName="Payer Space" appAbbr="PS" />
    );

    getByText('PS');
  });

  test('should add custom class name', () => {
    const { getByTestId } = render(
      <PageHeader appName="Payer Space" className="custom-classname" />
    );

    const pageHeader = getByTestId('page-header');

    expect(pageHeader.className).toContain('custom-classname');
  });

  test('should render app icon color', () => {
    const { getByTestId } = render(
      <PageHeader appName="Payer Space" appAbbr="PS" iconColor="green" />
    );

    const appIcon = getByTestId('page-header-app-icon');

    expect(appIcon.className).toContain('icon-green');
  });

  test('should render app icon color branded', () => {
    const { getByTestId } = render(
      <PageHeader
        appName="Payer Space"
        appAbbr="PS"
        iconColor="green"
        branded
      />
    );

    const appIcon = getByTestId('page-header-app-icon');

    expect(appIcon.className).toContain('icon-branded');
  });

  test('should render feedback', () => {
    const { getByTestId } = render(
      <PageHeader appName="Payer Space" feedback />
    );

    expect(getByTestId('face-options')).toBeDefined();
  });

  test('should render children', () => {
    const { getByText } = render(
      <PageHeader appName="Payer Space">this is cool</PageHeader>
    );

    getByText('this is cool');
  });

  test('should render trainingLink', () => {
    const { getByText } = render(
      <PageHeader
        appName="Payer Space"
        component={
          <TrainingLink
            name="Appeals"
            link="https://www.youtube.com/watch?v=GgwE94KZJ7E"
          />
        }
      >
        <p>this is cool</p>
      </PageHeader>
    );

    getByText('Watch a demo');
  });

  describe('spaces', () => {
    test('should work with spaceId', async () => {
      avSlotMachineApi.create.mockResolvedValue({
        data: {
          data: {
            spaces: {
              totalCount: 1,
              page: 1,
              perPage: 1,
              spaces: [{ id: '1', name: 'My Space' }],
            },
          },
        },
      });
      const { getByText } = render(
        <Spaces spaceIds={['1']} clientId="my-client-id">
          <PageHeader appName="Payer Space" spaceId="1" />
        </Spaces>
      );
      const defaultSpace = await waitForElement(() => getByText('My Space'));
      expect(defaultSpace.getAttribute('href')).toEqual(
        '/web/spaces/spaces/#/1'
      );
    });

    test('should work with payerId', async () => {
      avSlotMachineApi.create.mockResolvedValue({
        data: {
          data: {
            spaces: {
              totalCount: 1,
              page: 1,
              perPage: 1,
              spaces: [
                {
                  id: 'payer1',
                  name: 'My Space',
                  payerIDs: ['payer1'],

                  images: [
                    {
                      name: 'logo',
                      value: '/static/spaces/payer1/logo.png',
                    },
                  ],
                },
              ],
            },
          },
        },
      });
      const { getByTestId } = render(
        <Spaces payerIds={['payer1']} clientId="my-client-id">
          <PageHeader appName="Payer Space" payerId="payer1" />
        </Spaces>
      );

      await waitForElement(() => getByTestId('space-images.logo-payer1'));
    });
  });

  test('should render custom home url', () => {
    const { getByText } = render(
      <PageHeader homeUrl="/go-home" appName="Payer Space" />
    );

    const homeBtn = getByText('Home');

    expect(homeBtn.getAttribute('href')).toBe('/go-home');
  });

  test('should acccept custom title props', () => {
    const { getByTestId } = render(
      <PageHeader
        appName="Payer Space"
        titleProps={{
          className: 'mb-0',
        }}
      />
    );

    const pageHeaderTitle = getByTestId('page-header-title');

    expect(pageHeaderTitle.className).toContain('mb-0');
  });

  test('should render custom right content', () => {
    const { getByText } = render(
      <PageHeader
        appName="Payer Space"
        renderRightContent={({ feedback, ...props }) => (
          <div {...props}>
            Hello World
            {feedback}
          </div>
        )}
      />
    );

    getByText('Hello World');
    getByText('Give Feedback');
  });

  test('should render custom link tag', () => {
    const { getByText } = render(
      <PageHeader
        appName="Payer Space"
        renderRightContent={({ feedback, ...props }) => (
          <div {...props}>
            Hello World
            {feedback}
          </div>
        )}
        linkTag={NavLink}
      />
    );

    const homeBtn = getByText('Home');

    expect(homeBtn.className).toBe('nav-link');
  });

  test('should render with url', async () => {
    avSlotMachineApi.create.mockResolvedValue({
      data: {
        data: {
          spaces: {
            totalCount: 1,
            page: 1,
            perPage: 1,
            spaces: [
              { id: '1', name: 'My Space', link: { url: '/custom-link' } },
            ],
          },
        },
      },
    });
    const { getByText } = render(
      <Spaces spaceIds={['1']} clientId="my-client-id">
        <PageHeader appName="Payer Space" spaceId="1" />
      </Spaces>
    );

    const spaceBreadcrumb = await waitForElement(() => getByText('My Space'));

    expect(spaceBreadcrumb.tagName.toLowerCase()).toBe('a');

    expect(spaceBreadcrumb.getAttribute('href')).toEqual('/custom-link');
  });
});
