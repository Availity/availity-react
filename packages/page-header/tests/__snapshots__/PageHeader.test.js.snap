// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render 1`] = `
<div>
  <div
    class="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      class=""
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            aria-label="Home"
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
  </div>
  <h1
    class="page-header page-header-brand"
  >
    <div
      class="page-header-title"
    >
       
      Payer Space
    </div>
  </h1>
</div>
`;

exports[`PageHeader should render app icon 1`] = `
<div>
  <div
    class="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      class=""
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            aria-label="Home"
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
  </div>
  <h1
    class="page-header page-header-brand"
  >
    <div
      class="page-header-title"
    >
      <span
        class="app-icon app-icon-black"
        title="Payer Space"
      >
        PS
      </span>
       
      Payer Space
    </div>
  </h1>
</div>
`;

exports[`PageHeader should render app icon color 1`] = `
<div>
  <div
    class="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      class=""
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            aria-label="Home"
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
  </div>
  <h1
    class="page-header page-header-brand"
  >
    <div
      class="page-header-title"
    >
      <span
        class="app-icon app-icon-green"
        title="Payer Space"
      >
        PS
      </span>
       
      Payer Space
    </div>
  </h1>
</div>
`;

exports[`PageHeader should render app icon color branded 1`] = `
<div>
  <div
    class="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      class=""
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            aria-label="Home"
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
  </div>
  <h1
    class="page-header page-header-brand"
  >
    <div
      class="page-header-title"
    >
      <span
        class="app-icon app-icon-branded-green"
        title="Payer Space"
      >
        PS
        <span
          class="caret"
        />
      </span>
       
      Payer Space
    </div>
  </h1>
</div>
`;

exports[`PageHeader should render children 1`] = `
<div>
  <div
    class="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      class=""
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            aria-label="Home"
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
  </div>
  <h1
    class="page-header page-header-brand"
  >
    <div
      class="page-header-title"
    >
       
      <p>
        this is cool
      </p>
    </div>
  </h1>
</div>
`;

exports[`PageHeader should render feedback 1`] = `
<div>
  <div
    class="d-flex align-items-start"
  >
    <nav
      aria-label="breadcrumb"
      class=""
    >
      <ol
        class="breadcrumb"
      >
        <li
          class="breadcrumb-item"
        >
          <a
            aria-label="Home"
            href="/public/apps/dashboard"
          >
            Home
          </a>
        </li>
        <li
          aria-current="page"
          class="active breadcrumb-item"
        >
          Payer Space
        </li>
      </ol>
    </nav>
  </div>
  <h1
    class="page-header page-header-brand"
  >
    <div
      class="page-header-title"
    >
       
      Payer Space
    </div>
    <div
      class="float-md-right d-inline-block hidden-print dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="btn btn-light"
        type="button"
      >
        Give Feedback
      </button>
      <div
        aria-hidden="true"
        class="p-3 dropdown-menu dropdown-menu-right"
        role="menu"
        style="width: 400px;"
        tabindex="-1"
      >
        <div
          class="text-center"
        >
          <p>
            Tell us what you think about Payer Space.
          </p>
          <div
            class="btn-group"
            role="group"
          >
            <button
              class="btn btn-light"
              style="font-size: 1.4em; padding: .2em .4em;"
              type="button"
            >
              <span
                class="icon icon-smile"
              />
              <span
                class="sr-only"
              >
                Smiley face
              </span>
            </button>
            <button
              class="btn btn-light"
              style="font-size: 1.4em; padding: .2em .4em;"
              type="button"
            >
              <span
                class="icon icon-meh"
              />
              <span
                class="sr-only"
              >
                Meh face
              </span>
            </button>
            <button
              class="btn btn-light"
              style="font-size: 1.4em; padding: .2em .4em;"
              type="button"
            >
              <span
                class="icon icon-frown"
              />
              <span
                class="sr-only"
              >
                Frowny face
              </span>
            </button>
          </div>
          <div
            class="invalid-feedback"
          >
            Please select one of the faces above
          </div>
          <textarea
            aria-label="Please provide any feedback, requests, and/or defects."
            class="mt-3 form-control"
            placeholder="Feedback? Requests? Defects? (optional)"
          />
          <div
            class="mt-3 text-right"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Send
            </button>
          </div>
        </div>
      </div>
    </div>
  </h1>
</div>
`;
